includes/razorpay-subscriptions.php

```diff
@@ -113,6 +123,46 @@ protected function getSubscriptionCreateData($orderId)
         return $subscriptionData;
     }
 
+    protected function setStartAtIfNeeded(& $subscriptionData, & $signUpFee, $product)
+    {
+        $metadata = get_post_meta($product['product_id']);
+
+        if (empty($metadata['razorpay_wc_start_date']) === false)
+        {
+            //
+            // When custom start date is set, we consider the initial payment
+            // as a sign up payment, and the first recurring payment will be
+            // made on the configured start date on the next month
+            //
+            $startDay = $metadata['razorpay_wc_start_date'][0];
```

Comment:
- Handle error cases
  - What if date time doesn't parse correctly?
- Don't use explode/implode to parse dates, use datetime functions instead.
- We can't use `get_price`. See [this commit](https://github.com/razorpay/razorpay-woocommerce-subscriptions/commit/bf47f0d3166c051c08ac753165417184673f0a49)

Reply:
@captn3m0 Why won't date time parse correctly? We only set it if the product metadata contains a key we're looking for. The only way it would fail is if that is not in a valid range.

Checking for a valid number between 1 and 28 and throwing an exception accordingly. Let us tell merchants to ensure the date is between 1 and 28.